src/Uno.UI.RuntimeTests.Engine.Library/UI/UnitTestsControl.Filtering.cs

```diff
@@ -0,0 +1,81 @@
+﻿#if !UNO_RUNTIMETESTS_DISABLE_UI
+
+using System;
+using System.Collections.Generic;
+using System.Data;
+using System.Linq;
+using System.Reflection;
+using System.Text;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using static System.StringComparison;
+
+namespace Uno.UI.RuntimeTests;
+
+partial class UnitTestsControl
+{
+	private static IEnumerable<MethodInfo> FilterTests(UnitTestClassInfo testClassInfo, string? query)
+	{
+		var tests = testClassInfo.Tests?.AsEnumerable() ?? Array.Empty<MethodInfo>();
+		foreach (var filter in SearchPredicate.ParseQuery(query))
+		{
+			// chain filters with AND logic
+			tests = tests.Where(x =>
+				filter.Exclusion ^ // use xor to flip the result based on Exclusion
+				filter.Tag?.ToLowerInvariant() switch
+				{
+					"class" => x.DeclaringType?.Name.Contains(filter.Text, InvariantCultureIgnoreCase) ?? false,
+					// fixme: since we are returning MethodInfo for match, there is no way to specifying
+					// which of the [DataRow] or which row within [DynamicData] without refactoring.
+					"displayname" => x.GetCustomAttributes<DataRowAttribute>().Any(y => y.DisplayName.Contains(filter.Text, InvariantCultureIgnoreCase)),
```

Comment:
Could technically be a [DataRowAttribute](https://learn.microsoft.com/en-us/dotnet/api/microsoft.visualstudio.testtools.unittesting.datarowattribute?view=visualstudiosdk-2022) or a [DynamicDataAttribute](https://learn.microsoft.com/en-us/dotnet/api/microsoft.visualstudio.testtools.unittesting.dynamicdataattribute?view=visualstudiosdk-2022) (both implement [ITestDataSource](https://learn.microsoft.com/en-us/dotnet/api/microsoft.visualstudio.testtools.unittesting.itestdatasource?view=visualstudiosdk-2022))

Reply:
there is also method without said attribute...

Reply:
either that is a technicality... the big part is to refactor the up/downstream to make the filter happen later:
```
assembly.Classes.SelectMany(x => x.Methods).Where(filter...).SelectMany(x => x.DataRows ?? [x])
->
assembly.Classes.SelectMany(x => x.Methods).SelectMany(x => x.DataRows ?? [x]).Where(filter...)
```